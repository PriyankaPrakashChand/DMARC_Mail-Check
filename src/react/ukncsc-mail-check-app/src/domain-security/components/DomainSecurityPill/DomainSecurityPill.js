import React from 'react';
import { Label } from 'semantic-ui-react';
import PropTypes from 'prop-types';

import './DomainSecurityPill.css';

const getColor = (error, warning, information) =>
  (error && 'red') ||
  (warning && 'yellow') ||
  (information && 'grey') ||
  'green';

const DomainSecurityPill = ({ error, warning, information, children }) => (
  <Label circular color={getColor(error, warning, information)} size="large">
    <span className="DomainSecurityPill">{children}</span>
  </Label>
);

DomainSecurityPill.defaultProps = {
  error: false,
  warning: false,
  information: false,
};

DomainSecurityPill.propTypes = {
  error: PropTypes.bool,
  warning: PropTypes.bool,
  information: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default DomainSecurityPill;
