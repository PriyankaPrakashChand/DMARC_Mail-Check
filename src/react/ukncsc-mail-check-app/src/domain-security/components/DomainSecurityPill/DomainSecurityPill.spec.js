import React from 'react';
import { render } from 'react-testing-library';
import DomainSecurityPill from './DomainSecurityPill';

describe('DomainSecurityPill', () => {
  let container;

  describe('when getting an error pill', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityPill error>1 error!</DomainSecurityPill>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });

  describe('when getting a warning pill', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityPill warning>1 warning!</DomainSecurityPill>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });

  describe('when getting an information pill', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityPill error>1 info!</DomainSecurityPill>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });

  describe('when getting a success pill', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityPill>1 success!</DomainSecurityPill>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });
});
