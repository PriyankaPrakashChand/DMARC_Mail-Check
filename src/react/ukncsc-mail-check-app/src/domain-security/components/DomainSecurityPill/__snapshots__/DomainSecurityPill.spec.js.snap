// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainSecurityPill when getting a success pill it should match the snapshot 1`] = `
<div>
  <div
    class="ui green large circular label"
  >
    <span
      class="DomainSecurityPill"
    >
      1 success!
    </span>
  </div>
</div>
`;

exports[`DomainSecurityPill when getting a warning pill it should match the snapshot 1`] = `
<div>
  <div
    class="ui yellow large circular label"
  >
    <span
      class="DomainSecurityPill"
    >
      1 warning!
    </span>
  </div>
</div>
`;

exports[`DomainSecurityPill when getting an error pill it should match the snapshot 1`] = `
<div>
  <div
    class="ui red large circular label"
  >
    <span
      class="DomainSecurityPill"
    >
      1 error!
    </span>
  </div>
</div>
`;

exports[`DomainSecurityPill when getting an information pill it should match the snapshot 1`] = `
<div>
  <div
    class="ui red large circular label"
  >
    <span
      class="DomainSecurityPill"
    >
      1 info!
    </span>
  </div>
</div>
`;
